server/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -368,8 +368,15 @@ public T getDefault(Settings settings) {
     /**
      * Returns <code>true</code> iff this setting is present in the given settings object. Otherwise <code>false</code>
      */
-    public boolean exists(Settings settings) {
-        return settings.keySet().contains(getKey());
+    public boolean exists(final Settings settings) {
+        Setting<?> current = this;
```

Comment:
This method was not checking fallback settings, so we could get false here if the fallback setting was set but not the actual setting.